### PR TITLE
SONAR-6256 Replace component id columns by component uuid in dependencies table

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  public void execute(Context context) throws SQLException {
+    context.execute(generateSql());
+  }
+
+  private String generateSql() {
+    return new AddColumnsBuilder(db.getDialect(), "dependencies")
+      .addColumn(
+        new AddColumnsBuilder.ColumnDef()
+          .setName("from_component_uuid")
+          .setType(AddColumnsBuilder.ColumnDef.Type.STRING)
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java]: use import to have a shorter reference

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          .setLimit(50)
+          .setNullable(true)
+      )
+      .addColumn(
+        new AddColumnsBuilder.ColumnDef()
+          .setName("to_component_uuid")
+          .setType(AddColumnsBuilder.ColumnDef.Type.STRING)
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java]: use import to have a shorter reference

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/FeedDependenciesComponentUuids.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  public FeedDependenciesComponentUuids(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    MassUpdate update = context.prepareMassUpdate().rowPluralName("dependencies");
+    update.select(
+      "SELECT from_component.uuid, to_component.uuid, dependency.id " +
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/FeedDependenciesComponentUuids.java]: how will you handle case where component is not found in projects ? (partiel db cleaner)

File: server/sonar-server/src/test/java/org/sonar/server/db/migrations/v52/DropDependenciesComponentColumnsTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+public class DropDependenciesComponentColumnsTest {
+
+  DropDependenciesComponentColumns migration;
+
+  Database database;
+
+  @Before
+  public void setUp() throws Exception {
+    database = mock(Database.class);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/db/migrations/v52/DropDependenciesComponentColumnsTest.java]: what is missing in DbTester to use it to test column dropping ?

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java]: Why ?

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/FeedDependenciesComponentUuids.java (hunk 3 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/FeedDependenciesComponentUuids.java]: Indeed, but it should be taking into account by the task that will clean the db (if db could be "corrupted", it should be regularly clean up) -> I've added a comment in https://jira.codehaus.org/browse/SONAR-5503

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java (hunk 1 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java]: better readability ?

File: server/sonar-server/src/test/java/org/sonar/server/db/migrations/v52/DropDependenciesComponentColumnsTest.java (hunk 4 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/db/migrations/v52/DropDependenciesComponentColumnsTest.java]: Nothing, the issue is not on DbTester side but on H2 -> It does not support drop of multiple columns

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/AddDependenciesComponentUuidColumns.java]: That can be discussed, but why not

**Comment by teryk** [comment]
LGTM

**Pull request merged**